simulator/src/main/java/com/hedera/block/simulator/grpc/PublishStreamGrpcClientImpl.java

```diff
@@ -66,7 +72,7 @@ public void init() {
                 .usePlaintext()
                 .build();
         BlockStreamServiceGrpc.BlockStreamServiceStub stub = BlockStreamServiceGrpc.newStub(channel);
-        PublishStreamObserver publishStreamObserver = new PublishStreamObserver();
+        PublishStreamObserver publishStreamObserver = new PublishStreamObserver(streamEnabled);
```

Comment:
I do not think sharing this `streamEnabled` instance like that between `this` and the `publishStreamObserver` is a good idea. Yes, the state change will be visible, but think about how error prone could that be. Without the context as to what this variable intends to do, in the future it will be misused. Private members _must_ remain so and interaction with the outside world should not happen like that. This variable is not updated anywhere in this class, it is only used to read from, but updated in another place. That being said, this should really be a property of the place that is responsible for updating it.

It would be better if the `streamEnabled` is a private member of the `PublishStreamObserver` and to be read via a method call from that object like `publisStreamObserver.isStreamEnabled()` rather than shared between two places and read directly from the instance on line 104 of this class.

Reply:
I think we're in an in-between state at the moment where all DI options will look a bit ugly.
Injecting the observer requires a lot of extra setup that shouldn't be externalized.
Injecting the boolean looks a bit ugly, but _should_ be a temporary expedient.

Longer term I hope we'll collect this and several other items into some sort of stream control object that makes more sense and looks much cleaner when injected.

Reply:
@jsync-swirlds I agree with the in-between state, we also discussed that with @georgi-l95 as well. For me his changes are ok based on that fact, my proposal aims to clean up a bit and make easier to test, and I am aware of the fact that this is absolutely not a more permanent production ready solution. Also, yes, lets discuss in the future what the approach should be, not only about the control flow, but error handling approach in general.

@georgi-l95 as we spoke, the proposed is up to you if you see fit to implement, I agree with both either way, since with the proposed we achieve _not_ much more than maybe a bit better testing for now, but maybe you can consider only to not share the atomic boolean but poll it via a method as shown in [this comment](https://github.com/hashgraph/hedera-block-node/pull/327#discussion_r1824547888) since this change is very small, you simply delete the field from the client and call a method instead (the atomic boolean remains as a constructor parameter for the client to be able to pass the tests).

Reply:
Let's go atm with the approach where we introduce this variable via injection in constructor. It's a hack, but will do the job for now. We will 100% revisit and refactor this whole flow in the future.

Reply:
@georgi-l95 agreed on my side.